docs/aqua-book/language/crdt-maps.md

```diff
@@ -0,0 +1,81 @@
+# CRDT Maps
+
+A map is a kind of [collection](types.md#collection-types) that store arrays of elements that can be accessible by `string` keys. 
+
+```aqua
+-- A map creation
+map: %u64
+
+-- Write to a map, `foo()` returns `u64`
+map <- "key", 1234
+map <- "key", foo()
+map <- "second key", foo()
+```
+
+Access to keys and elements can create different behaviours based on tasks:
+
+```aqua
+map: %u64
+
+-- Get arrays of elements ([]u64) by key in place
+values <- map.get("key")
+
+-- Get stream of elements (*u64) by key, later you can join on number of elements and wait for specific elements in this stream
+valuesStream <- map.getStream("key")
+-- ...
+join valuesStream[10]
+
+-- Get array of keys ([]string) in place
+keys <- map.keys()
+
+-- Get stream of keys (*string)
+keysStream <- map.keysStream
+
+-- Check if key contains at least one element, return boolean
+keyExist <- map.contains("key")
+
+```
+
+`get`, `keys` and `contains` methods will get non-changed result in-place, on the other hand `getStream` and `keysStream` methods will get streams, so, these streams can be changed over time while a map is updated during the execution. For example:
```

Comment:
What is "non-changed" in this context? You mean immutable?
"Can be changed over time" is vague too. It reads like they can be changed by the developer, but the important thing is that they are eventually updated with keys added to the map.